controllers/article-controllers.js

```diff
@@ -0,0 +1,24 @@
+const { articleById } = require("../models/article-models");
+
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+
+  if (!/^\d+$/.test(article_id)) {
+    const error = new Error("Bad Request");
+    error.msg = "Bad request";
+    error.status = 400;
+    return next(error);
```

Comment:
return custom errors in model using promise rejections please.

Reply:
you do not need to throw custom error for bad request as psql will throw error for wrong data type for the id, which you'll need to catch in a different error handler (see later comment for more).